code/modules/mob/living/silicon/robot/robot_mob.dm

```diff
@@ -447,6 +449,84 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 
 	return module_sprites
 
+/**
+  * Sets the offset for a cyborg's hats based on their module icon.
+  *
+  * Arguments:
+  * * module - An `icon_state` for which the offset needs to be calculated.
+  */
+/mob/living/silicon/robot/proc/robot_module_hat_offset(module)
+	switch(module)
+		if("Engineering", "Miner_old", "JanBot2", "Medbot", "engineerrobot", "maximillion", "secborg", "Hydrobot")
+			can_be_hatted = FALSE
+			hat_offset_y = -1
+		if("Noble-CLN", "Noble-SRV", "Noble-DIG", "Noble-MED", "Noble-SEC", "Noble-ENG", "Noble-STD")
+			can_be_hatted = TRUE
+			can_wear_restricted_hats = TRUE
+			hat_offset_y = 4
+		if("droid-medical")
+			can_be_hatted = TRUE
+			can_wear_restricted_hats = TRUE
+			hat_offset_y = 4
```

Comment:
Are you separating them by borg categories instead of sharing values?

Reply:
They're grouped by cyborg sprite type, in this example all the Noble sprites are the same borg sprites but recoloured. Can change the grouping to just have all in the same bracket if they share values though, if preferred.

Reply:
Either way works, if kept i would just leave a comment here about it being grouped by recoloured borg sprites

Reply:
Gone ahead and added a comment